cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDocumentStoreTemplate.java

```diff
@@ -65,6 +60,57 @@ public void initialize(final AmazonDynamoDB amazonDynamoDbClient) {
         dynamoDBClient = new DynamoDB(amazonDynamoDbClient);
     }
 
+    /**
+     * Simple method for splitting a list into a list of smaller lists of the supplied length
+     * 
+     * @param list
+     * @param length
+     * @return
+     */
+    private <T> List<List<T>> split(final List<T> list, final int length) {
+        final List<List<T>> parts = new ArrayList<List<T>>();
+        final int size = list.size();
+        for (int i = 0; i < size; i += length) {
+            parts.add(new ArrayList<T>(list.subList(i, Math.min(size, i + length))));
+        }
+        return parts;
+    }
+
+    private <T extends Item> List<T> executeQuery(final KeySetQuery query, final Class<T> itemClass) {
+
+        final ItemConfiguration itemConfiguration = getItemConfiguration(itemClass);
+        final String tableName = databaseSchemaHolder.schemaName() + "." + itemConfiguration.tableName();
+
+        // max 100 keys per fetch so split in to lists of lists
+        final List<List<ItemId>> split_ids = split(new ArrayList<ItemId>(query.itemIds()), 100);
+        final List<T> fetchedItems = new ArrayList<T>();
+
+        for (final List<ItemId> ids : split_ids) {
+            final TableKeysAndAttributes keys = new TableKeysAndAttributes(tableName);
+            for (final ItemId id : ids) {
+                keys.addPrimaryKey(getPrimaryKey(id, itemConfiguration));
+            }
+            BatchGetItemOutcome outcome = dynamoDBClient.batchGetItem(keys);
+            do {
+                final List<com.amazonaws.services.dynamodbv2.document.Item> items = outcome.getTableItems().get(
+                        tableName);
+                for (final com.amazonaws.services.dynamodbv2.document.Item item : items) {
+                    fetchedItems.add(stringToItem(item.toJSON(), itemClass));
+                }
+                // Check for unprocessed keys which could happen if you exceed provisioned
+                // throughput or reach the limit on response size.
+                final Map<String, KeysAndAttributes> unprocessedKeys = outcome.getUnprocessedKeys();
+                if (outcome.getUnprocessedKeys().size() == 0) {
+                    logger.debug("No unprocessed keys found");
+                } else {
+                    logger.debug("Retrieving the unprocessed keys");
+                    outcome = dynamoDBClient.batchGetItemUnprocessed(unprocessedKeys);
```

Comment:
FYI this is 99.9% AWS example code

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/batch-operation-document-api-java.html#JavaDocumentAPIBatchGet